packages/EasyDoctrine/src/Subscribers/EntityEventSubscriber.php

```diff
@@ -138,20 +155,27 @@ public function onFlush(OnFlushEventArgs $eventArgs): void
      *
      * @return array<string, array<mixed>>
      */
-    public function computeCollectionsChangeSet(
+    private function computeCollectionsChangeSet(
         array $collections,
         EntityManagerInterface $entityManager,
     ): array {
         $changeSet = [];
-        $mappingIdsFunction = static function (object $entity) use ($entityManager): string {
-            $identifierName = \current($entityManager->getClassMetadata(\get_class($entity))->getIdentifier());
-            return (string)$entityManager->getUnitOfWork()
-                ->getEntityIdentifier($entity)[$identifierName];
+        $mappingIdsFunction = function (object $entity) use ($entityManager): mixed {
+            $id = $entityManager->getUnitOfWork()
+                ->getSingleIdentifierValue($entity);
+
+            if ($id === null) {
+                $id = $this->deletedEntitiesIds[\spl_object_hash($entity)];
```

Comment:
There is no guarantee that we have this item in the array. Even if we think this should never happen, it would be better to add at least `?? null`.